IRremoteESP8266.cpp

```diff
@@ -713,26 +713,35 @@ void ICACHE_FLASH_ATTR IRsend::space(unsigned long time) {
   }
 }
 
+// Calculate the period for a given frequency. (T = 1/f)
+//
+// Args:
+//   freq: Frequency in Hz.
+// Returns:
+//   nr. of uSeconds.
+uint32_t ICACHE_FLASH_ATTR IRsend::calcUSecPeriod(uint32_t hz) {
+  return (1000000UL + hz/2) / hz;  // round(1000000/hz).
```

Comment:
No, it's not for hardware lag, it's just to get a better integer representation of the 'fraction'.
I can add references for round(), but you're not going to like it. :)
e.g. https://en.wikipedia.org/wiki/Half-integer, https://en.wikipedia.org/wiki/Nearest_integer_function, & https://en.wikipedia.org/wiki/Rounding#Rounding_to_integer

It's a horribly complex topic.
Previously we were doing the equiv. of a pure trunc(). e.g. trunc(1.9) == 1, trunc(2.1) == 2, etc.
In order to give us a round() rather than a straight truncation, adding 0.5 (to a positive integer), then trunc()'ing gives us a _good enough\<tm pat. pend>_ round() implementation.
e.g. trunc(1.1 + 0.5) == trunc(1.6) == 1, where as trunc(1.6 + 0.5) == trunc(2.1) == 2.

As for why `hz/2`, (hz/2)/hz (taking in more of the expression) is the equiv of 1/2, as the hz's cancel out. i.e. hz/hz == 1.

Full form, you'd get
(1000000 + hz/2)/hz == 1000000/hz + hz/2/hz == 1000000/hz + 1/2
&
round(x) == trunc(x + 1/2) (see previous example for that)

Now, If you can sum that up in a few lines, or find a nice link that works (and isn't this comment ;-), I'm happy to use that. Otherwise, this is a tricky thing to explain in a few lines of comments (I think).